### ISSUE TITLE
Allow passing extra options to CodeNarc when using shared library

### ISSUE THREAD

**Issue by maxdanilov** [AbletonDevTools/groovylint#97]
Currently, shared library only allows docker image invocation passing default parameters:
https://github.com/AbletonDevTools/groovylint/blob/master/vars/groovylint.groovy
```groovy
  image.inside("-v ${env.WORKSPACE}:/ws") {
    sh "python3 /opt/run_codenarc.py -- -includes=${includesPattern}"
  }
```

For the production use, it would be nice to have at least the possibility to use a custom rules file like this:
```groovy
  python3 /opt/run_codenarc.py -- -includes='*.groovy' -rulesetfiles=file:myrules.groovy
```
Same applies to the ability to specify the output type (xml, text, html etc.)

Could be something as simple as passing the extra params to the method:
```groovy
void check(String includesPattern, Object groovylintImage = null, String extraParams = '') {
// ...
  image.inside("-v ${env.WORKSPACE}:/ws") {
    sh "python3 /opt/run_codenarc.py -- -includes=${includesPattern} ${extraParams}"
  }
```

**Comment by nre-ableton** [comment]
Sure, this works for me. See https://github.com/AbletonDevTools/groovylint/pull/98.

**Comment by maxdanilov** [comment]
@nre-ableton 
Awesome, thank you 🙇

**Comment by nre-ableton** [comment]
Fixed in groovylint 0.7.1.
